src/odemis/gui/comp/viewport.py

```diff
@@ -2208,8 +2208,13 @@ def connect_stream(self, projs):
         stream, proj = self._stream, self._projection
 
         if hasattr(stream, "selected_angle") and hasattr(stream, "selected_wavelength"):
+            self.canvas.add_view_overlay(self.canvas.markline_overlay)
             pos = self.canvas.markline_overlay.val
             pos.value = (stream.selected_wavelength.value, math.degrees(stream.selected_angle.value))  # (nm, degrees)
+        else:
+            # Hide the line overlay if not useful. Especially, for the raw EK data
+            # the angles list contains NaNs which the overlay doesn't support.
+            self.canvas.remove_view_overlay(self.canvas.markline_overlay)
```

Comment:
It is nice to add something like this here:
`logging.debug("stream %s has no selected angle and/or selected_wavelength the view overlay is removed")`